libraries/standard/mqtt/src/mqtt.c

```diff
@@ -428,7 +434,7 @@ static bool matchWildcards( const char * pTopicName,
                             const char * pTopicFilter,
                             uint16_t topicFilterLength,
                             uint16_t * pNameIndex,
-                            uint16_t filterIndex,
+                            uint16_t * pFilterIndex,
```

Comment:
Updated `filterIndex` to be an output parameter to be an output parameter to address corner case of matching first level between topic filter and topic name (when topic filter starts with `+`  and topic name has empty first level as  `/`) that resulted in underflow when calling `( *pNameIndex ) --` when input `*pNameIndex` is 0. 

By making `filterIndex` an output parameter, instead of decrementing the `*pNameIndex` value, we increment the `filterIndex` value and thereby proceed the indices of both topic name and topic filter in matching to avoid the underflow problem of the corner case

Reply:
Added.